### ISSUE TITLE
afterLoad event not correct

### ISSUE THREAD

**Issue by saaseasyllc** [typeorm/typeorm#6989]
afterLoad event is not a LoadEvent, it is the entity itself....

event.entity
event.metadata are undefined.

These are correctly populated in afterInsert however, but not in afterLoad

I notice InsertEvent, RemoveEvent, and UpdateEvent are imported from "typeorm" , while LoadEvent is imported from a different location, which seems odd.

typeorm: tried 0.2.25 and 0.2.28

```
import { EntityMetadata, EntitySubscriberInterface, EventSubscriber, InsertEvent, RemoveEvent, UpdateEvent } from "typeorm";
import { LoadEvent } from "typeorm/subscriber/event/LoadEvent";
import { SEEventBus, SEEvent } from "../../../../eventbus/se.event.bus";
import { SELogger } from "../../../../logger/se.logger";

const CN: string = "TypeORMSubscriber";

@EventSubscriber()
export class TypeORMSubscriber implements EntitySubscriberInterface<any> {
.....
    afterLoad(event: LoadEvent<any>) {
        let M: string = "afterLoad()";
        SELogger.debug(CN+M+'AFTER LOADED: ' + JSON.stringify(event));
        if(!event){
            SELogger.error(CN+M+" event not defined");
            return;
        }
       //event is not a LoadEvent, it is the entity

        let metadata: EntityMetadata = event.metadata;
       //undefined!
       
        let seEvent: SEEvent = new SEEvent();
        seEvent.resource = event;
        //@ts-ignore
        seEvent.resourceType = event.constructor.name;
        seEvent.type = "resourceLoaded";
        SELogger.debug(CN+M+" event: ");
        SELogger.debug(JSON.stringify(seEvent));
        SEEventBus.publish("resourceLoaded", seEvent);
    }
....
}
```
`

**Comment by sbleon** [comment]
For historical reasons, `afterLoad` works differently than the other events. The signature is:
`afterLoad(entity: any, event: LoadEvent<any>)`.

This is documented at https://typeorm.io/#/listeners-and-subscribers/what-is-a-subscriber

I think this issue can be closed.

**Comment by mjdikun** [comment]
I tried the signature you suggested, however I don't think that is the signature either...  not getting the event: LoadEvent

I think it's just afterLoad(entity:any)

I am having an issue with afterUpdate as well, but at this point i'm just going to put my event publishing into a different layer...

Log: 

query: UPDATE `employees` SET `deletedAt` = CURRENT_TIMESTAMP, `updatedAt` = CURRENT_TIMESTAMP WHERE `id` IN (?) -- PARAMETERS: ["bb234727-5a51-40f9-ab07-6e334e6205b9"]
debug: TypeORMSubscriberafterUpdate()&&&&&&&&&&&&&&&&&&&&&&&&&&&&AFTER UPDATE: undefined
error: TypeORMSubscriberafterUpdate() event.entity not defined

```
 afterUpdate(event:UpdateEvent<any>){
        let M: string = "afterUpdate()";
        SELogger.debug(CN+M+`&&&&&&&&&&&&&&&&&&&&&&&&&&&&AFTER UPDATE: ` + JSON.stringify(event, getCircularReplacer));
        if(!event.entity){
            SELogger.error(CN+M+" event.entity not defined");
            return;
        }
        let metadata: EntityMetadata = event.metadata;
        SELogger.debug(M+metadata.name+"");
        let seEvent: SEEvent = new SEEvent();
        seEvent.resource = event.entity;
        seEvent.resourceType = metadata.name;
        seEvent.type = "resourceUpdated";
        SEEventBus.publish("resourceUpdated", seEvent);
    }
```

**Comment by sbleon** [comment]
That's very strange! I'm also using v0.2.28, and I get an `Entity` and a `LoadEvent` as args to my `afterLoad`. I'm importing `LoadEvent` from the same place that you are.

I would have suspected you're using a (very much) older version of typeorm, because the sig for `afterLoad` used to be `afterLoad(Entity)` before v0.2.18. You can see more about this here: https://github.com/typeorm/typeorm/blob/master/src/subscriber/EntitySubscriberInterface.ts#L20-L28 .

**Comment by mjdikun** [comment]
Yep, I see that signature too when I got to the EntitySubscriberInterface definition in my editor, however event is undefined in afterLoad, I do notice it is optional as it's marked with event?: ...

and no idea why afterUpdate(event) is undefined completely....

I've been reading some other issues where events are dependent on the Typeorm CRUD Api's called... which is not good for my use case, I need events to be dependable, so that's why i'm going to event publishing to my Repository level above Typeorm.
